sql/updates/latest-dev.sql

```diff
@@ -1,2 +1,88 @@
 DROP FUNCTION _timescaledb_internal.time_col_name_for_chunk(name,name);
 DROP FUNCTION _timescaledb_internal.time_col_type_for_chunk(name,name);
+
+-- Handle column renames for continuous aggregates that were not
+-- handled correctly and fix it in the update. We save the information
+-- in a table.
+CREATE TABLE rename_tables (
+       user_view regclass,
+       new_name text,
+       old_name text,
+       partial_view regclass,
+       direct_view regclass,
+       mat_table regclass,
+       hypertable_id int
+);
+
+-- Compare the user view and the direct view of each continuous
+-- aggregate to figure out what columns that were renamed on the user
+-- view but which did not propagate to the other objects of the
+-- continuous aggregate since this did not work in previous versions.
+WITH
+  objs AS (
+        SELECT format('%I.%I', user_view_schema, user_view_name)::regclass AS user_view,
+               format('%I.%I', direct_view_schema, direct_view_name)::regclass AS direct_view,
+               format('%I.%I', partial_view_schema, partial_view_name)::regclass AS partial_view,
+               format('%I.%I', schema_name, table_name)::regclass AS mat_table,
+               mat_hypertable_id AS mat_id
+          FROM _timescaledb_catalog.continuous_agg
+          JOIN _timescaledb_catalog.hypertable ON mat_hypertable_id = id),
+  user_view AS (
+        SELECT attrelid, attname, attnum, mat_id
+          FROM objs, pg_attribute
+         WHERE attrelid = objs.user_view),
+  direct_view AS (
+        SELECT attrelid, attname, attnum, mat_id
+          FROM objs, pg_attribute
+         WHERE attrelid = objs.direct_view)
+INSERT INTO rename_tables
+SELECT (SELECT user_view FROM objs WHERE uv.attrelid = user_view),
+       uv.attname AS new_name,
+       dv.attname AS old_name,
+       (SELECT partial_view FROM objs WHERE uv.attrelid = user_view),
+       (SELECT direct_view FROM objs WHERE uv.attrelid = user_view),
+       (SELECT mat_table FROM objs WHERE uv.attrelid = user_view),
+       (SELECT mat_id FROM objs WHERE uv.attrelid = user_view)
+  FROM user_view uv JOIN direct_view dv USING (mat_id, attnum)
+ WHERE uv.attname != dv.attname;
+
+CREATE PROCEDURE alter_table_column(cagg_name text, table_name text, old_name text, new_name text) AS $$
```

Comment:
why is that a procedure and not a function

Reply:
No specific reason, but it made more sense to use a procedure since it does not need to return anything and only contains DDL. Why would it be a function?